tests/test_swift.py

```diff
@@ -29,5 +78,100 @@ def test_frames(self):
+            self.buses.update(appInfo.bus)
+        self.swift = Swift(APP_INFOS)
+
+    def test_init(self):
+        """Test if swift is initialized correctly."""
+        self.assertIsInstance(self.swift.mainWindow, QMainWindow)
+        self.assertIsInstance(self.swift.centralWidget, QLabel)
+        for name in APP_INFOS:
+            self.assertIn(name, self.swift._apps)
+        for bus in self.buses:
+            self.assertIn(bus, self.swift._subscribers)
+
+    def test_destroy_app(self):
+        """Test destroyApp()."""
+        name = list(APP_INFOS.keys())[0]
+        self.swift.destroyApp(name)
+        self.assertNotIn(name, self.swift._apps)
+
+    def test_broadcast(self):
+        """Test _broadcast()."""
+        busName = list(self.buses)[0]
+        self.swift._broadcast(busName, "test_msg")
+        for app in self.swift._subscribers[busName]:
+            app.received.emit.assert_called_once()
+
+
+class SwiftFunctionTest(unittest.TestCase):
+    """Unit test for functions in swift.py"""
+
+    def test_parse(self):
+        """Test parse()."""
+        self.assertEqual(parse(AppInfo, APP_JSONS["app1"]), APP_INFOS["app1"])
+        self.assertEqual(parse(AppInfo, APP_JSONS["app2_default"]), APP_INFOS["app2"])
+
+    def test_strinfo(self):
+        """Test strinfo()."""
+        self.assertEqual(strinfo(APP_INFOS["app1"]), APP_JSONS["app1"])
+        self.assertEqual(strinfo(APP_INFOS["app2"]), APP_JSONS["app2"])
+
+    def test_add_to_path(self):
+        """Test _add_to_path()."""
+        old_path = sys.path.copy()
+        with _add_to_path(os.path.dirname("test_dir")):
+            self.assertNotEqual(old_path, sys.path)
+        self.assertEqual(old_path, sys.path)
+
+    @patch.object(sys, "argv", ["", "-s", "test_setup.json"])
+    def test_get_argparser(self):
+        """Test _get_argparser()."""
+        parser = _get_argparser()
+        args = parser.parse_args()
+        self.assertEqual(parser.description, "SNU widget integration framework for PyQt")
+        self.assertEqual(args.setup_path, "test_setup.json")
+
+    @patch.object(sys, "argv", [""])
+    def test_get_argparser_default(self):
+        """Test _get_argparser() with default options."""
+        args = _get_argparser().parse_args()
+        self.assertEqual(args.setup_path, "./setup.json")
+
+    @patch("builtins.open")
+    @patch("json.load", return_value={"app": APP_DICTS})
+    def test_read_setup_file(self, mock_open, mock_load):
+        """Test _read_setup_file()."""
+        self.assertEqual(_read_setup_file(""), APP_INFOS)
+        mock_open.assert_called_once()
+        mock_load.assert_called_once()
+
+    @patch("swift.swift._get_argparser")
+    @patch("swift.swift._read_setup_file", return_value=({}, {}))
+    @patch("swift.swift.Swift")
+    @patch("PyQt5.QtWidgets.QApplication.exec_")
+    def test_main(self, mock_get_argparser, mock_read_setup_file, mock_swift, mock_exec_):
+        """Test main()."""
```

Comment:
Hmm.. the docstrings seems too trivial.
Should we omit the docstring for testing unless there is additional context should be noted?
Please give me your opinion.

Reply:
Thank you for the suggestions.
I would prefer the second option, with the pylintrc for testing.